sui/src/config/mod.rs

```diff
@@ -222,16 +249,31 @@ impl NetworkConfig {
                         .public_key
                         .make_narwhal_public_key()
                         .expect("Can't get narwhal public key");
+
+                    debug!("Resolving {}:{} for {}", &x.host, &x.port, name);
+
                     let primary = PrimaryAddresses {
-                        primary_to_primary: socket_addr_from_hostport(&x.host, x.port + 100),
-                        worker_to_primary: socket_addr_from_hostport(&x.host, x.port + 200),
+                        primary_to_primary: socket_addr_from_hostport_retry(&x.host, x.port + 100),
```

Comment:
Why don't we just replace the original function and always retry?  I'm not sure there's a case when we don't want to retry for this.

Reply:
This function is used in other places that may bring unexpected side effects for them. e.g. on deser: https://github.com/MystenLabs/sui/blob/main/sui/src/config/mod.rs#L148